internal/sigstore/verify.go

```diff
@@ -12,37 +12,50 @@ import (
 	"encoding/base64"
 	"fmt"
 
-	"github.com/edgelesssys/constellation/v2/internal/api/versionsapi"
 	"github.com/edgelesssys/constellation/v2/internal/constants"
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
 	sigsig "github.com/sigstore/sigstore/pkg/signature"
 )
 
 // Verifier checks if the signature of content can be verified.
 type Verifier interface {
-	VerifySignature(content, signature, publicKey []byte) error
+	VerifySignature(content, signature []byte) error
 }
 
-// CosignVerifier checks if the signature of content can be verified
-// using a cosign public key.
-type CosignVerifier struct{}
-
-// VerifySignature checks if the signature of content can be verified
-// using publicKey.
-// signature is expected to be base64 encoded.
-// publicKey is expected to be PEM encoded.
-func (CosignVerifier) VerifySignature(content, signature, publicKey []byte) error {
-	sigRaw := base64.NewDecoder(base64.StdEncoding, bytes.NewReader(signature))
+// CosignVerifier wraps a public key that can be used for verifying signatures.
+type CosignVerifier struct {
+	publicKey crypto.PublicKey
+}
 
-	pubKeyRaw, err := cryptoutils.UnmarshalPEMToPublicKey(publicKey)
+// SetPublicKey parses the pem encoded public key, validates it's parameters and updates the object if necessary.
+func (c *CosignVerifier) SetPublicKey(pem []byte) error {
+	pubkey, err := cryptoutils.UnmarshalPEMToPublicKey(pem)
```

Comment:
Why have this as a separate function instead of using a constructor which takes a public key as argument?
The `CosignVerifier` does not work without this function being called first, so it would make sense to use it as a constructor for it, no?
Additionally I currently don't see any use cases where we would want to change the used public key during a verification process, so this function should only ever be called once.

Reply:
I agree, setters are always dangerous when they need to be called in order for some other function to work.